eox_core/tests/integration/test_enrollment_integration.py

```diff
@@ -0,0 +1,610 @@
+                                     self.data['endpoint'])
+        response = requests.post(request_url,
+                                 data=data,
+                                 headers=headers)
+        self.assertEqual(response.status_code, 400)
+
+    def test_create_valid_user_mode_invalid_course_for_site(self):
+        # pylint: disable=invalid-name
+        """
+        Create enrollment with a valid user, valid mode,
+        and a course from another site
+        """
+        site1_data = self.data['site1_data']
+        site2_data = self.data['site2_data']
+        data = {
+            'email': site2_data['user_email'],
+            'course_id': site1_data['course']['id'],
+            'mode': site1_data['course']['mode'],
+            'force': True,
+        }
+        headers = {
+            'Authorization': 'Bearer {}'.format(site2_data['token']),
+            'Host': site2_data['host'],
+        }
+        request_url = '{}/{}'.format(site2_data['base_url'],
+                                     self.data['endpoint'])
+        response = requests.post(request_url,
+                                 data=data,
+                                 headers=headers)
+        self.assertEqual(response.status_code, 400)
+
+    # NOTE: Mode changes are not working correctly on devstack
+    def test_force_create_valid_user_course_invalid_mode(self):
+        # pylint: disable=invalid-name
+        """
+        Create enrollment with a valid user, valid course,
+        and a not available mode
+        """
+        delete_enrollment(self.data)
+
+        site1_data = self.data['site1_data']
+        data = {
+            'email': site1_data['user_email'],
+            'course_id': site1_data['course']['id'],
+            'mode': 'masters',
+            'force': 1,
+        }
+        headers = {
+            'Authorization': 'Bearer {}'.format(site1_data['token']),
+            'Host': site1_data['host'],
+        }
+        request_url = '{}/{}'.format(site1_data['base_url'],
+                                     self.data['endpoint'])
+        response = requests.post(request_url,
+                                 data=data,
+                                 headers=headers)
+        self.assertEqual(response.status_code, 400)
+
+    def test_delete_valid_enrollment(self):
+        # pylint: disable=invalid-name
+        """
+        Delete a valid enrollment
+        """
+        create_enrollment(self.data)
+        site1_data = self.data['site1_data']
+        data = {
+            'email': site1_data['user_email'],
+            'course_id': site1_data['course']['id'],
+        }
+        headers = {
+            'Authorization': 'Bearer {}'.format(site1_data['token']),
+            'Host': site1_data['host'],
+        }
+        request_url = '{}/{}'.format(site1_data['base_url'],
+                                     self.data['endpoint'])
+        response = requests.delete(request_url,
```

Comment:
Maybe it will look better if the requests have this format
```
        response = requests.delete(
            request_url,
            data=data,
            headers=headers,
        )
```

Reply:
Since [we are probably going to use](https://github.com/eduNEXT/eox-core/pull/126#pullrequestreview-565378735) an auto-formatting tool in the future, I've decided to just run black on the file and go with that format.

Reply:
Looks good. I think the only thing missing is the trailing comma in the requests